example/generator/src/SinewaveTelemetryProvider.js

```diff
@@ -24,96 +24,47 @@
 /**
  * Module defining SinewaveTelemetryProvider. Created by vwoeltje on 11/12/14.
  */
-define(
-    ["./SinewaveTelemetrySeries"],
-    function (SinewaveTelemetrySeries) {
-        "use strict";
+define([
+    "./SinewaveTelemetrySeries",
+    "./GeneratorProvider"
+], function (
+    SinewaveTelemetrySeries,
+    GeneratorProvider
+) {
 
-        /**
-         *
-         * @constructor
-         */
-        function SinewaveTelemetryProvider($q, $timeout) {
-            var subscriptions = [],
-                generating = false;
-
-            //
-            function matchesSource(request) {
-                return request.source === "generator";
-            }
-
-            // Used internally; this will be repacked by doPackage
-            function generateData(request) {
-                return {
-                    key: request.key,
-                    telemetry: new SinewaveTelemetrySeries(request)
-                };
-            }
-
-            //
-            function doPackage(results) {
-                var packaged = {};
-                results.forEach(function (result) {
-                    packaged[result.key] = result.telemetry;
-                });
-                // Format as expected (sources -> keys -> telemetry)
-                return { generator: packaged };
-            }
-
-            function requestTelemetry(requests) {
-                return $timeout(function () {
-                    return doPackage(requests.filter(matchesSource).map(generateData));
-                }, 0);
-            }
-
-            function handleSubscriptions() {
-                subscriptions.forEach(function (subscription) {
-                    var requests = subscription.requests;
-                    subscription.callback(doPackage(
-                        requests.filter(matchesSource).map(generateData)
-                    ));
-                });
-            }
-
-            function startGenerating() {
-                generating = true;
-                $timeout(function () {
-                    handleSubscriptions();
-                    if (generating && subscriptions.length > 0) {
-                        startGenerating();
-                    } else {
-                        generating = false;
-                    }
-                }, 1000);
-            }
+    function SinewaveTelemetryProvider() {
```

Comment:
Having both an old and new provider will result in duplicate telemetry from the new API when the legacy adapter in open1348 is merged. We can do one of -
1. Add an adapter going back the other way (to expose new style providers to old views using the old API) and remove the SineWaveProvider here
2. Update our remaining views to use the new telemetry API, and drop the SineWaveProvider

Since we want to do 2. anyway, we should probably just prioritize that. We may need a temporary stopgap to merge #1348 though (such as temporarily disabling the registration of the GeneratorProvider).